plugins/uefi-capsule/fu-uefi-cod-device.c

```diff
@@ -173,13 +173,26 @@ fu_uefi_cod_device_get_indexed_filename(FuUefiDevice *self, GError **error)
 static gchar *
 fu_uefi_cod_device_get_filename(FuUefiDevice *self, GError **error)
 {
+	FuContext *ctx = fu_device_get_context(FU_DEVICE(self));
 	g_autofree gchar *esp_path = fu_volume_get_mount_point(fu_uefi_device_get_esp(self));
 	g_autofree gchar *basename = NULL;
+	const gchar *manufacturer = NULL;
 
 	/* InsydeH2O */
 	if (fu_device_has_private_flag(FU_DEVICE(self), FU_UEFI_DEVICE_FLAG_COD_INDEXED_FILENAME))
 		return fu_uefi_cod_device_get_indexed_filename(self, error);
 
+	/* Dell Inc. */
+	manufacturer = fu_context_get_hwid_value(ctx, FU_HWIDS_KEY_MANUFACTURER);
+	if (g_strcmp0(manufacturer, "Dell Inc.") == 0)
+		return g_build_filename(esp_path,
+					"EFI",
+					"dell",
+					"bios",
+					"recovery",
+					"BIOS_TRS.rcv",
+					NULL);
```

Comment:
What happens if there are two capsule-on-disk updates scheduled? Or is that no possible?

Reply:
I think fwupd replaces the old file with the newer one.

Reply:
I'm a bit confused why it's called `bios/recovery` -- is this really the path you want to use for normal capsule upgrades?

Reply:
> What happens if there are two capsule-on-disk updates scheduled? Or is that no possible?

As an idea to cover this case I think it's better to set a private flag earlier in the plugin startup and only set CoD for the "System Firmware" device.  Let any other UEFI devices be set to use NVRAM.

Reply:
> only set CoD for the "System Firmware" device 

That's good idea - I have a growing feeling that this path is essentially designed for BIOS Recovery somewhat is being reused for this purpose.  We probably don't want to deploy capsules other than BIOS f/w.

Reply:
I also wonder if your BIOS team has given you the wrong info; it seems quite a divergence from the UEFI specification!

Reply:
> from the UEFI specification

Customized path for now, will change back to spec mentioned path when available.

Reply:
Did the BIOS team say what they were going to implement? I'm wondering how we're supposed to know what path to use going forwards.

Reply:
e.g. I'm wondering if doing something like fwupd/plugins/uefi-capsule/uefi-capsule.quirk, where we have:

    # LENOVO
    [6de5d951-d755-576b-bd09-c5cf66b27234]
    Flags = supports-boot-order-lock,use-legacy-bootmgr-desc,no-ux-capsule,no-lid-closed,modify-bootorder

i.e. we'd have something like a plugin flag of `use-dell-recovery` for the HWIDs of the systems affected?

Reply:
I was looking at the quirk too, wanted to add `cod-bios-only` likewaise but device kind (e.g.: kind system firmware) is known at the uefi-device lifecycle however current implementation put gtype at backend time which is earlier than uefi-device.  Any ideas?

I was thinking to create both types and inhibit/hidden the unused gtype created devices.

Reply:
or use cod for all uefi-capsule devices regardless system firmware or other kinds because bios is the only target at this time.

Reply:
There is no longer microcode or ssd device firmware this way?

Reply:
no restriction on system-firmware deviece, spread to all devices with `org.uefi.capsule` plugin. The path will be changed to the customized one.